types/src/v0/impls/auction.rs

```diff
@@ -0,0 +1,274 @@
+use crate::{
+    eth_signature_key::{EthKeyPair, SigningError},
+    v0_1::ValidatedState,
+    v0_3::{BidTx, BidTxBody, FullNetworkTx},
+    FeeAccount, FeeAmount, FeeError, FeeInfo, NamespaceId,
+};
+use committable::{Commitment, Committable};
+use ethers::types::Signature;
+use hotshot_types::{
+    data::ViewNumber,
+    traits::{
+        auction_results_provider::HasUrl, node_implementation::ConsensusTime,
+        signature_key::BuilderSignatureKey,
+    },
+};
+use std::str::FromStr;
+use thiserror::Error;
+use url::Url;
+
+impl FullNetworkTx {
+    /// Proxy for `execute` method of each transaction variant.
+    pub fn execute(&self, state: &mut ValidatedState) -> Result<(), ExecutionError> {
+        match self {
+            Self::Bid(bid) => bid.execute(state),
+        }
+    }
+}
+
+impl Committable for BidTxBody {
+    fn tag() -> String {
+        "BID_TX".to_string()
+    }
+
+    fn commit(&self) -> Commitment<Self> {
+        let comm = committable::RawCommitmentBuilder::new(&Self::tag())
+            .fixed_size_field("account", &self.account.to_fixed_bytes())
+            .fixed_size_field("gas_price", &self.gas_price.to_fixed_bytes())
+            .fixed_size_field("bid_amount", &self.bid_amount.to_fixed_bytes())
+            .var_size_field("url", self.url.as_str().as_ref())
+            .u64_field("view", self.view.u64())
+            .var_size_field("namespaces", &bincode::serialize(&self.namespaces).unwrap());
+        comm.finalize()
+    }
+}
+
+impl BidTxBody {
+    /// Construct a new `BidTxBody`.
+    pub fn new(
+        account: FeeAccount,
+        bid: FeeAmount,
+        view: ViewNumber,
+        namespaces: Vec<NamespaceId>,
+        url: Url,
+    ) -> Self {
+        Self {
+            account,
+            bid_amount: bid,
+            view,
+            namespaces,
+            url,
+            // TODO gas_price will come from config probably, but we
+            // can use any value for first round of integration
+            ..Self::default()
+        }
+    }
+
+    /// Sign `BidTxBody` and return the signature.
+    pub fn sign(&self, key: &EthKeyPair) -> Result<Signature, SigningError> {
```

Comment:
Do we expect to use this function outside of this crate? If not, can it be a private function, or removed (i.e., `signed` would call `sign_builder_message` directly), to make it clearer that to get `BidTx` we just need to use `signed`?

Reply:
Yea, I guess my preference is in early integration is to have too much rather than not enough. df255e6a813d8e403076e00ae573a46108baf2ef